### PR TITLE
feat: consider auto save for documents

### PR DESCRIPTION
**Pull request by cguedes**
Auto-save documents

- [x] remove dirty indicator from tab
- [x] create utility hook to check if any file is dirty
- [x] debounced save of editor content after update
- [x] footer item indicator

https://github.com/refstudio/refstudio/assets/174127/15a97306-8e48-4156-a490-ec2cd950c666

This is related to #484

### CODE REVIEW

File: src/atoms/hooks/__tests__/useAnyEditorDataIsDirty.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { waitFor } from '@testing-library/react';
```
Comment (sehyod) [inline src/atoms/hooks/__tests__/useAnyEditorDataIsDirty.test.ts:1]: You should import it from `'../../../test/test-utils';`

File: src/atoms/hooks/__tests__/useAnyEditorDataIsDirty.test.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  it('should return true when one editor is edited', async () => {
+    const hookResult = runHookWithJotaiProvider(useAnyEditorDataIsDirty, store);
+
+    expect(hookResult.current).toBeFalsy();
+
+    store.set(initialEditorsContent.get(editorId1)!.updateEditorContentBufferAtom, {
+      type: 'refstudio',
+      jsonContent: { doc: 'Updated content' },
+    });
```
Comment (sehyod) [inline src/atoms/hooks/__tests__/useAnyEditorDataIsDirty.test.ts:47]: You should to wrap this in `act`

**Review by sehyod — CHANGES_REQUESTED** [review]

File: src/test/test-utils.tsx [hunk 3]
```diff
@@ -21,7 +21,9 @@ function customRender(ui: React.ReactElement, options = {}) {
   });
 }
 
-export * from '@testing-library/react';
+// export * from '@testing-library/react';
```
Comment (sehyod) [inline src/test/test-utils.tsx:24]: Should we remove this line?

**Review by sehyod — APPROVED** [review]

File: src/test/test-utils.tsx (hunk 3 above)
Comment (cguedes) [reply src/test/test-utils.tsx:24]: Sure that was a "typo"

**Pull request merged**